### PR TITLE
fix missing chinese characters when using ttf label

### PR DESCRIPTION
**Pull request by zilongshanren**
https://github.com/cocos2d/cocos2d-x/issues/16784

### CODE REVIEW

**Comment by dumganhar** [comment]
👍  Good finding.

File: cocos/platform/desktop/CCGLViewImpl-desktop.cpp [hunk 1]
```diff
@@ -911,6 +911,10 @@ void GLViewImpl::onGLFWframebuffersize(GLFWwindow* window, int w, int h)
         _retinaFactor = 1;
         glfwSetWindowSize(window, static_cast<int>(frameSizeW * _retinaFactor * _frameZoomFactor), static_cast<int>(frameSizeH * _retinaFactor * _frameZoomFactor));
     }
+
+    //FIXME:https://github.com/cocos2d/cocos2d-x/issues/16779
```
Comment (zilongshanren) [inline cocos/platform/desktop/CCGLViewImpl-desktop.cpp:915]: This  fix is for  moving GLFW window from one monitor to another.

**Comment by zilongshanren** [comment]
The second commmit for:  https://github.com/cocos2d/cocos2d-x/issues/16779

File: cocos/platform/desktop/CCGLViewImpl-desktop.cpp (hunk 1 above)
Comment (zilongshanren) [reply cocos/platform/desktop/CCGLViewImpl-desktop.cpp:915]: @dumganhar  please take a look at this commit, thanks.

File: cocos/platform/desktop/CCGLViewImpl-desktop.cpp [hunk 2]
```diff
@@ -911,6 +911,10 @@ void GLViewImpl::onGLFWframebuffersize(GLFWwindow* window, int w, int h)
...
+    Director::getInstance()->setViewport();
+    Director::getInstance()->mainLoop();
```
Comment (dumganhar) [inline cocos/platform/desktop/CCGLViewImpl-desktop.cpp:917]: Why should invoke mainLoop after setViewpoint?, mainLoop will be automatically called, right?

**Comment by dumganhar** [comment]
@zilongshanren , could you please pull the latest code and resolve the conflicts?

**Comment by zilongshanren** [comment]
@dumganhar  I split this PR into two:

https://github.com/cocos2d/cocos2d-x/pull/16832

https://github.com/cocos2d/cocos2d-x/pull/16833

**Comment by Monisha123Ramesh** [comment]
hello @zilongshanren, @dumganhar
I am doing a game for Mac store using cocos2d-js. I am in need of help with edit box issue.
Edit box works weird in Mac. When the window size is changed (i.e.,maximised or minimised) the edit box position changes and it comes back to its original position until its been clicked. I am sharing the screenshots. Can you please have a check on it and provide me a solution as I'm stuck on this issue for long..


Initial:

   screen shot 2017-12-28 at 3 07 51 pm

After maximising

screen shot 2017-12-28 at 3 08 18 pm

Thank you.

**Pull request closed without merging**